### PR TITLE
Add PCOV to TravisCI

### PR DESCRIPTION
**Pull request by maks-rafalko**
* Now, instead of `$PHPDBG`, `$NO_DEBUGGER` etc. there is a`$DRIVER` env variable that can be `xdebug`, `phpdbg`, `pcov`, `none`. Took idea from [PHPUnit's travis config](https://github.com/sebastianbergmann/php-code-coverage/pull/663/files)
* Remove duplicate jobs from travis (example: https://travis-ci.org/infection/infection/builds/555199408, see two first jobs in the `Test` stage)
* Do not run `pcov` for phpspec e2e test, it does not work
* Use `composer install` or `composer update --prefer-lowest` in our `E2ETest` that runs e2e tests inside PHPUnit. This fixes the issue when we run job on Travis with `$DEPS="LOW"`, and Infection itself has the lowest dependencies (e.g. PHPUnit `8.0.0`), but e2e test has normal `^8` version wich is resolved to `8.2.4` atm. Error was: [`PHP Fatal error:  Uncaught Error: Call to undefined method SebastianBergmann\Environment\Runtime::getCurrentSettings() in /home/travis/build/infection/infection/tests/Fixtures/e2e/Save_PHPUnit_Bootstrap_File/vendor/phpunit/phpunit/src/Util/PHP/AbstractPhpProcess.php:191`](https://travis-ci.org/infection/infection/jobs/556475734#L573)
* Remove mutators form `PhpUnit_Depends_Order` e2e test. See explanation inside the diff

### Current testing matrix:

|  PHP | Driver  | Dependencies |
|:-:|:-:|:-:|
| 7.2  | `xdebug`  | |
| 7.2  | `phpdbg`  | |
| 7.2  | `pcov` (NEW)  | |
| 7.2  | `none`  | |
| 7.2  | `xdebug`  | lowest |
| 7.2  | `phpdbg`  | lowest |
| 7.3  | `xdebug`  | |
| 7.3  | `phpdbg`  | |
| 7.3  | `none`  | |
| 7.3  | `xdebug`  | lowest |
| 7.3  | `phpdbg`  | lowest |
| 7.4snapshot  | `phpdbg`  |  |
| nightly  | `phpdbg`  |  |

I'm not sure we need `pcov` for 7.3 and lowest dependencies. All of this increases a build time a lot.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,20 +1,20 @@
-!/.tools/.gitkeep
-!/build/.gitkeep
+/build/
```
[the remark this answers is not in the document]
Comment (maks-rafalko) [reply .gitignore]: yes

File: .travis.yml [hunk 2]
```diff
@@ -70,60 +83,57 @@ jobs:
...
       <<: *STANDARD_TEST_JOB
       php: 7.3
-      env: DEPS="LOW"
+      env: DEPS="LOW" DRIVER="xdebug"
 
     -
       <<: *STANDARD_TEST_JOB
       php: 7.3
-      env: DEPS="LOW" PHPDBG=1
+      env: DEPS="LOW" DRIVER="phpdbg"
```
[the remark this answers is not in the document]
Comment (maks-rafalko) [reply .travis.yml]: I'm ok with reducing the matrix

**Comment by sanmai** [comment]
> lowest dependencies (e.g. PHPUnit 8.0.0)

Shall we limit the lowest version to something more sensible? Like 8.2.4 which is the most current release now.

**Comment by sanmai** [comment]
I think one way to tackle this issue is to detect whenever we're being run with `DEPS="LOW"`, and then just skip the `E2ETest`.

**Comment by maks-rafalko** [comment]
@theofidry, @sanmai fixed your comments.

New matrix

Build #2753 - infection:infection - Travis CI 2019-07-12 22-34-32

File: .travis.yml [hunk 3]
```diff
@@ -41,24 +41,37 @@ jobs:
...
+                phpize
+                ./configure
+                make clean install
+                echo "extension=pcov.so" > $HOME/.phpenv/versions/$TRAVIS_PHP_VERSION/etc/conf.d/pcov.ini
+                cd $TRAVIS_BUILD_DIR
+              fi
       script:
-        - if [[ $PHPDBG != 1 ]]; then xdebug-enable; fi
-        - if [[ $PHPDBG != 1 ]]; then $PHPUNIT_BIN; else phpdbg -qrr $PHPUNIT_BIN; fi
+        - if [[ "$DRIVER" == "xdebug" || "$DRIVER" == "none" ]]; then xdebug-enable; fi
```
Comment (sanmai) [inline .travis.yml]: So the point of `DRIVER=none` is to run Infection without any kind of coverage collector, with the coverage data we had from the initial PHPUnit run, am I right?

Comment (maks-rafalko) [reply .travis.yml]: Exactly!

**Comment by maks-rafalko** [comment]
Thank you @sanmai and @theofidry for the review

**Pull request merged**